beacon-chain/node/node.go

```diff
@@ -118,10 +118,14 @@ func New(cliCtx *cli.Context, opts ...Option) (*BeaconNode, error) {
 		return nil, err
 	}
 	prereqs.WarnIfPlatformNotSupported(cliCtx.Context)
-	features.ConfigureBeaconChain(cliCtx)
+	if err := features.ConfigureBeaconChain(cliCtx); err != nil {
+		return nil, err
+	}
 	cmd.ConfigureBeaconChain(cliCtx)
 	flags.ConfigureGlobalFlags(cliCtx)
-	configureChainConfig(cliCtx)
+	if err := configureChainConfig(cliCtx); err != nil {
```

Comment:
this hints at one of the big reasons for this PR - previously if a value was given to OverrideBeaconConfig that would cause a fork version conflict in the registry, we would have to panic because there weren't defined error handling paths. As an incremental step, I kept the behavior of OverrideBeaconConfig (swapping pointers without updating the registry) but ensured that paths where we are taking a config file as user input (like here in the beacon-chain node initialization routines), we use some new loader methods that return errors, which will bubble up to the CLI.